src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageAccountProvider.cs

```diff
@@ -169,14 +168,21 @@ private IStorageAccount StorageAccount
                 if (!string.IsNullOrEmpty(connectionString))
                 {
                     account = ParseAccount(connectionStringName, connectionString);
-                    isPrimary = false;
                 }
             }
 
             if (account != null)
             {
                 // On the first attempt, this will make a network call to verify the credentials work.
-                await _storageCredentialsValidator.ValidateCredentialsAsync(account, isPrimary, cancellationToken);
+                await _storageCredentialsValidator.ValidateCredentialsAsync(account, cancellationToken);
+
+                if (!supportedTypes.HasFlag(account.Type))
```

Comment:
How about we move this out into its own static helper method, since the logic is separate from validating/classifying the account. E.g. perhaps an extension method to IStorageAccount.

That way, you don't have to modify all of the existing function signatures to pass in the type. I think there are only 2 places where you're actually doing an account restriction check anyways - so you can call to the new method in those places only.